core/api/src/modules/stateMachine.ts

```diff
@@ -0,0 +1,54 @@
+import { Model } from "sequelize-typescript";
+
+export namespace StateMachine {
+  export interface StateTransition {
+    from: string;
+    to: string;
+    checks: Array<string>;
+  }
+
+  export async function transition(
+    instance: Model,
+    transitions: Array<StateTransition>
+  ) {
+    const klass = modelName(instance);
+    const newState: string = instance["state"];
+    //@ts-ignore
+    const oldState: string = instance._previousDataValues["state"];
+    if (!newState || !oldState || newState === oldState) {
```

Comment:
i'm worried `!oldState` here misses initial validation (running `checks` on creation. Is that true?